keras_nlp/tokenizers/sentence_piece_tokenizer.py

```diff
@@ -169,6 +169,11 @@ def get_vocabulary(self) -> List[str]:
 
     def id_to_token(self, id: int) -> str:
         """Convert an integer id to a string token."""
+        if id >= self.vocabulary_size() or id < 0:
+            raise ValueError(
+                f"`id` must be in range [0, {self.vocabulary_size()-1}], "
```

Comment:
Looks good! This would be slightly more in line with our style (perhaps inconsistently enforced).

```
                f"`id` must be in range [0, {self.vocabulary_size() - 1}]. "
                f"Recieved: {id}"
```